### PR TITLE
Allow setup of Mosquito Add-on in MQTT config flow

### PR DESCRIPTION
**Pull request by jbouwh**
## Proposed change

Allow setup of Mosquitoo Add-on in MQTT config flow

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/124106
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Enhanced documentation for setting up MQTT integration with Home Assistant.
	- Introduced automated configuration for the Mosquitto Broker add-on, including secure username and password generation.
	- Added details on creating extra logins for MQTT devices via the Mosquitto add-on configuration.
	- Mentioned the option for users to independently configure a different MQTT broker for added flexibility.

### CODE REVIEW

File: source/_integrations/mqtt.markdown [hunk 1]
```diff
@@ -119,6 +119,11 @@ MQTT (aka MQ Telemetry Transport) is a machine-to-machine or "Internet of Things
 
 Your first step to get MQTT and Home Assistant working is to choose a broker.
 
+The easiest option is to install the official Mosquitto Broker add-on, you can choose to set up and configure this add-on automatically when you set up the MQTT integration. Home Assistant will automatically generate and assign a safe username and password and further attention is required. This also works if you have already set up this add-on yourself in advance.
```
Comment (c0ffeeca7) [inline source/_integrations/mqtt.markdown]:
```suggestion
The easiest option is to install the official Mosquitto Broker add-on. You can choose to set up and configure this add-on automatically when you set up the MQTT integration. Home Assistant will automatically generate and assign a safe username and password, and further attention is required. This also works if you have already set up this add-on yourself in advance.
```

File: source/_integrations/mqtt.markdown [hunk 2]
```diff
@@ -119,6 +119,11 @@ MQTT (aka MQ Telemetry Transport) is a machine-to-machine or "Internet of Things
 
 Your first step to get MQTT and Home Assistant working is to choose a broker.
 
+The easiest option is to install the official Mosquitto Broker add-on. You can choose to set up and configure this add-on automatically when you set up the MQTT integration. Home Assistant will automatically generate and assign a safe username and password, and further attention is required. This also works if you have already set up this add-on yourself in advance.
```
Comment (c0ffeeca7) [inline source/_integrations/mqtt.markdown]: should this be "and no further attention is required"?

Comment (jbouwh) [reply source/_integrations/mqtt.markdown]: Nice catch! I'll change it!

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @jbouwh :+1:

**Pull request merged**